src/components/RangeSelector.js

```diff
@@ -130,20 +172,30 @@ class RangeSelector extends React.Component {
       newState[this.state.dragging.toLowerCase() + 'Pixels'] = newPixels;
       newState[this.state.dragging.toLowerCase() + 'Value'] = newValue;
 
-      this.setState(newState);
+      this.setState(newState, () => {
+        this.props['on' + this.state.dragging + 'DragStop'](this.state[this.state.dragging.toLowerCase() + 'Value']);
+
+        if (this.state.trackClicked) {
+          this.setState({
+            dragging: false,
+            trackClicked: false
+          });
+        }
+      });
```

Comment:
I tried this, and the logs you see below are logging the previous value of the upper and lower toggle when the track is clicked - is that an issue? 

```
	  if (this.state.trackClicked) {
  		newState.dragging = false;
  		newState.trackClicked = false;
	  }
	  
	  //Logs previous value of the toggle, when track is clicked
	  console.log("this is this.state.upperValue", this.state.upperValue); 
	  console.log("this is this.state.lowerValue", this.state.lowerValue);
	  
	  this.props['on' + this.state.dragging + 'DragStop'](newState[this.state.dragging.toLowerCase() + 'Value']);

	  this.setState(newState);
```
<img width="1412" alt="screen shot 2016-01-14 at 11 24 59 am" src="https://cloud.githubusercontent.com/assets/13579578/12333439/79a3dd66-bab1-11e5-8310-154e7a42eb0f.png">